plugins/plugin-github/codenvy-plugin-github-factory-resolver/src/main/java/com/codenvy/plugin/factory/github/factory/resolver/GithubFactoryParametersResolver.java

```diff
@@ -0,0 +1,186 @@
+
+import static java.util.Collections.singletonList;
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+/**
+ * Provides Factory Parameters resolver for github repositories.
+ *
+ * @author Florent Benoit
+ */
+public class GithubFactoryParametersResolver implements FactoryParametersResolver {
+
+    /**
+     * Logger.
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(GithubFactoryParametersResolver.class);
+
+    /**
+     * Parameter name.
+     */
+    protected static final String URL_PARAMETER_NAME = "url";
+
+    /**
+     * Connection timeout of 10seconds.
+     */
+    private static final int CONNECTION_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(10);
+
+    /**
+     * Regexp to find repository details (repository name, project name and branch and subfolder)
+     * Examples of valid URLs are in the test class.
+     */
+    private static final Pattern GITHUB_PATTERN = Pattern.compile("^(?:http)(?:s)?(?:\\:\\/\\/)github.com/(?<repoUser>[^/]++)/(?<repoName>[^/]++)(?:/tree/(?<branchName>[^/]++)(?:/(?<subFolder>[^/]++))?)?$");
+
+    /**
+     * Check if this resolver can be used with the given parameters.
+     *
+     * @param factoryParameters
+     *         map of parameters dedicated to factories
+     * @return true if it will be accepted by the resolver implementation or false if it is not accepted
+     */
+    @Override
+    public boolean accept(@NotNull Map<String, String> factoryParameters) {
+        // Check if url parameter is a github URL
+        return factoryParameters.containsKey(URL_PARAMETER_NAME) && GITHUB_PATTERN.matcher(factoryParameters.get(URL_PARAMETER_NAME)).matches();
+    }
+
+    /**
+     * Create factory object based on provided parameters
+     *
+     * @param factoryParameters
+     *         map containing factory data parameters provided through URL
+     * @throws BadRequestException when data are invalid
+     */
+    @Override
+    public Factory createFactory(@NotNull Map<String, String> factoryParameters) throws BadRequestException {
+
+        // no need to check null value of url parameter as accept() method has performed the check
+        String githubUrl = factoryParameters.get("url");
+
+        // Apply github url to the regexp
+        Matcher matcher = GITHUB_PATTERN.matcher(githubUrl);
+        if (!matcher.matches()) {
+            throw new BadRequestException(String.format(
+                    "The given github url %s is not a valid URL github url. It should start with https://github.com/<user>/<repo>",
+                    githubUrl));
+        }
+
+        String repoUser = matcher.group("repoUser");
+        String repoName = matcher.group("repoName");
+        String branchName = matcher.group("branchName");
+        String subfolder = matcher.group("subFolder");
+
+        // check required parameter are there
+        if (repoUser == null && repoName == null) {
+            throw new BadRequestException(String.format(
```

Comment:
As far as I understand this clause is not needed because of check that url matches the regexp.